operators/src/adapters/raster_subquery/raster_subquery_reprojection.rs

```diff
@@ -306,34 +306,38 @@ where
     } = accu;
 
     let mut materialized_accu_tile = accu_tile.into_materialized_tile(); //in a fold chain the real materialization should only happen once. All other calls will be simple conversions.
+    let axis_size_x = materialized_accu_tile.grid_array.shape.axis_size_x();
+    let axis_size_y = materialized_accu_tile.grid_array.shape.axis_size_y();
 
     pool.install(|| {
         let parallelism = pool.current_num_threads();
-        let par_chunk_split = num::integer::div_ceil(
-            materialized_accu_tile.grid_array.shape.axis_size_y(),
-            parallelism,
-        )
-        .max(min_rows_in_par_chunk); // don't go below MIN_ROWS_IN_PAR_CHUNK lines per chunk.
-        let par_chunk_size =
-            materialized_accu_tile.grid_array.shape.axis_size_x() * par_chunk_split;
+        let par_chunk_split =
+            num::integer::div_ceil(axis_size_y, parallelism).max(min_rows_in_par_chunk); // don't go below MIN_ROWS_IN_PAR_CHUNK lines per chunk.
+        let par_chunk_size = axis_size_x * par_chunk_split;
 
         materialized_accu_tile
             .grid_array
             .data
             .par_chunks_mut(par_chunk_size)
             .enumerate()
-            .for_each(|(y_f, row_slice)| {
-                let y = y_f * par_chunk_split;
-                row_slice.iter_mut().enumerate().for_each(|(x, pixel)| {
-                    let lookup_coord = coords
-                        .get_at_grid_index_unchecked(GridIdx2D::from([y as isize, x as isize]));
-                    if let Some(coord) = lookup_coord {
-                        if tile.spatial_partition().contains_coordinate(&coord) {
-                            let lookup_value = tile.pixel_value_at_coord_unchecked(coord);
-                            *pixel = lookup_value;
+            .for_each(|(chunk_idx, chunk_slice)| {
+                let chunk_start_y = chunk_idx * par_chunk_split;
+                chunk_slice
+                    .iter_mut()
+                    .enumerate()
+                    .for_each(|(lin_idx, pixel)| {
+                        let x_idx = lin_idx % axis_size_x;
+                        let y_idx = lin_idx / axis_size_x + chunk_start_y;
```

Comment:
you calculate the `y_idx` for each pixel. Isn't the way I do it in the interploation operator more efficient? (Calculating start and end y and then iterating over slices with precalculated y index)

Also: can you please add a test that would have failed without this change. It worries me that this change doesn't break any tests...

Reply:
i added debug_asserts so the error in question should show up in testing